### ISSUE TITLE
[BUG] Duplicate tracks in album view, Directories view is correct

### ISSUE THREAD

**Issue by bbsixzz** [BLeeEZ/amperfy#110]
iPhone SE 2016 iOS 15.4.1, Amperfy 1.1.0 (1)

Certain albums have incorrect number of tracks in Album view but appear correct in Directories view.

They appear correctly in Navidrome and iSub, I've also tried to reset app data to no avail.

FabricLive 25: High Contrast [[XML](https://www.toptal.com/developers/hastebin/oquqahorak.lua)] has 22 tracks but incorrectly shows 26.

https://user-images.githubusercontent.com/47334163/162585478-c17eafe1-881e-4e12-bd99-acdfff332a6d.MP4

https://user-images.githubusercontent.com/47334163/162585476-24d25b8e-c90a-481b-862c-45dff05a74c8.MP4

Be Up a Hello [[XML](https://www.toptal.com/developers/hastebin/rikaluwodi.lua)] has 9 tracks but shows 11.

![IMG_5685](https://user-images.githubusercontent.com/47334163/162587946-63760953-5be1-432d-82b7-4effd8b57cec.PNG)

![IMG_5684](https://user-images.githubusercontent.com/47334163/162587943-74f789c4-8d29-41e5-8428-34f52af91f4f.PNG)

**Comment by bbsixzz** [comment]
After trying to reset app data again it kept crashing while doing so, I've sent the bug reports to you.

I then removed the app and reinstalled only to find that the High Contradt album now reports 35 tracks in album view!

![9B254CD7-5F57-4B46-A6EE-D8F81101B828](https://user-images.githubusercontent.com/47334163/162619784-b7799199-aa7c-4953-a37d-0af9f67ce2e4.png)

**Comment by BLeeEZ** [comment]
I have to remove the app data reset function. There are some problems. Yes if the app crashes during reset there can definitely be problems with partially deleted library elements. Can you please report if this problem rises again and app data reset has not been used? This issue needs to be fixed if there is an other root cause.

**Comment by bbsixzz** [comment]
Yes I will keep checking, this app is now my daily driver since the AirPlay fix.

**Comment by BLeeEZ** [comment]
Nice to hear.
"Reset App Data" is not available anymore with this commit: ce84279ee3605f40e5199ea89ad973ec1a54585b

**Comment by bbsixzz** [comment]
It happened again.

I had these files as FLAC on the server, I deleted them and replaced them with MP3.

After that it seems to leave duplicate entries in Album view, Directories view is fine.

![IMG_E6AE9759E806-1](https://user-images.githubusercontent.com/47334163/163266983-8f4e72df-05dc-478d-87eb-bea82a8c051a.jpeg)

![IMG_40E4F12943C4-1](https://user-images.githubusercontent.com/47334163/163266989-98299c11-bbfd-47b8-b85b-ecd876f3dd52.jpeg)

**Comment by bbsixzz** [comment]
The missing files have greyed out artist name and produce this error now:

![IMG_5CEDF5DF8644-1](https://user-images.githubusercontent.com/47334163/163278738-e828ab31-316d-4fdb-84f3-0fb2faa354c4.jpeg)

**Comment by bbsixzz** [comment]
I've discovered more duplicates, the files on the server have not been tampered with in this instance so there is no API Error.

The playback will play the tracks twice.

![IMG_A666FC178325-1](https://user-images.githubusercontent.com/47334163/163420654-c1488b02-9147-43ce-9f2e-7df94d87327b.jpeg)

**Comment by BLeeEZ** [comment]
Small update on this issue . I found two bugs that both result in the issue described above.

The first is that remote deleted files are not getting deleted the Amperfy library when album details are getting fetched in. This is fixed with 303b6a3a2969f12689efefc7a6f0f1a9320c822b. Small remark: if the remote deleted file has been cached in Amperfy it will be still displayed. It hides immediately if the song is remove from cache.

The other bug is that in some cases songs/albums/artists get created twice (race conditions). This bug will be fixed next.

**Comment by BLeeEZ** [comment]
With this commit 95276c92eea0dc7be225faa65df48b91cddca1ed duplicated entities will be merged. This issue should now be fixed.

**Comment by BLeeEZ** [comment]
Is this bug fixed on your device?

**Comment by bbsixzz** [comment]
I think this is fixed, thank you!
